### PR TITLE
refactor: directional light do not illuminate the primitives behind it

### PR DESCRIPTION
**Pull request by mathieurobert1**
Before with the light placed between the two spheres:
image
Now with the refactor:
image

### CODE REVIEW

**Review by AxelF44 — DISMISSED** [review]

File: include/Math/Vector3D.hpp [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 #pragma once
 
+#define IS_INVERSE(X,Y) (((X > 0 && Y < 0) || (X < 0 && Y > 0)) ? true : false)
```
Comment (Njord201) [inline include/Math/Vector3D.hpp:10]: Change the name of the Macro to IS_ON_REAR

File: src/Plugins/Lights/Directional/Directional.cpp [hunk 2]
```diff
@@ -65,6 +65,11 @@ Math::Point3D Light::Directional::computeColor(Math::Vector3D primitiveNormal, c
     Math::Vector3D pr = primitiveNormal / primitiveNormal.length();
     Math::Vector3D dir = getDirection() / getDirection().length();
 
+    Math::Vector3D hit = hitPoint;
+    Math::Vector3D rayOriginToHit = hit - _position;
+    if (IS_INVERSE(rayOriginToHit.x(), dir.x()) || IS_INVERSE(rayOriginToHit.y(), dir.y()) || IS_INVERSE(rayOriginToHit.z(), dir.z()))
+        return Math::Point3D(0,0,0);
```
Comment (Njord201) [inline src/Plugins/Lights/Directional/Directional.cpp:71]: Use VOID_COLOR Macro

**Review by Njord201 — CHANGES_REQUESTED** [review]

File: include/Math/Vector3D.hpp (hunk 1 above)
Comment (mathieurobert1) [reply include/Math/Vector3D.hpp:10]: Not agree

**Review by theophile-jr — CHANGES_REQUESTED** [review]
conflicts & requests

File: src/Plugins/Lights/Directional/Directional.cpp [hunk 3]
```diff
@@ -6,6 +6,7 @@
 */
 
 #include "Lights/Directional.hpp"
+#include "RaytracerRules.hpp"
```
Comment (AxelF44) [inline src/Plugins/Lights/Directional/Directional.cpp]:
```suggestion
#include "RaytracerRules.hpp"
#include "Lights/Directional.hpp"
```

**Review by AxelF44 — APPROVED** [review]

**Review by Njord201 — APPROVED** [review]

**Review by AxelF44 — APPROVED** [review]

**Review by Njord201 — APPROVED** [review]

**Review by theophile-jr — COMMENTED** [review]
PR NAME : ENGLISH ISSUE

**Review by theophile-jr — APPROVED** [review]

**Pull request merged**